src/main/java/core/mate/academy/impl/TruckProducer.java

```diff
@@ -0,0 +1,19 @@
+package core.mate.academy.impl;
+
+import core.mate.academy.model.Truck;
+import core.mate.academy.service.MachineProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {
+    private List<Truck> list = new ArrayList<>();
```

Comment:
The field 'list' should have an access modifier to encapsulate the internal representation. Consider using 'private' to ensure that the list cannot be accessed directly from outside the class.